pkg/alertmanager/operator.go

```diff
@@ -804,15 +804,21 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	sErr, ok := err.(*apierrors.StatusError)
 
 	if ok && sErr.ErrStatus.Code == 422 && sErr.ErrStatus.Reason == metav1.StatusReasonInvalid {
-		c.metrics.StsDeleteCreateCounter().Inc()
-
 		// Gather only reason for failed update
 		failMsg := make([]string, len(sErr.ErrStatus.Details.Causes))
 		for i, cause := range sErr.ErrStatus.Details.Causes {
 			failMsg[i] = cause.Message
 		}
 
-		level.Info(logger).Log("msg", "recreating AlertManager StatefulSet because the update operation wasn't possible", "reason", strings.Join(failMsg, ", "))
+		level.Error(logger).Log("msg", "updating StatefulSet failed", "reason", strings.Join(failMsg, ", "))
```

Comment:
I wouldn't log an error here: users will be worried while the operator handles the case.